.github/workflows/test_fast.yml

```
@@ -104,9 +77,47 @@ jobs:
           path: coverage.xml
           retention-days: 7
 
+  lint:
+    runs-on: 'ubuntu-latest'
+    timeout-minutes: 10
+    steps:
+      - name: Apt-Get Install
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y shellcheck
+
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      # note: exclude python 3.10+ from mypy checks as these produce false
+      # positives in installed libraries for python 3.7
+      - name: Configure Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+
+      - name: Install
+        run: |
+          pip install -e ."[tests]"
```

Oops, linkcheck is actually failing because of missing dependencies by the looks of it
```suggestion
          pip install -e ."[all]"
```

---

It shouldn't need cylc-flow installed.

In the output it's failed whilst trying to import `tests/unit/main_loop/test_log_db.py` which is very odd.